### PR TITLE
Fix reloading for 'latest' extension versions

### LINKED ISSUE #25751: AMP page is breaking when browser cached files are loading
AMP page is breaking when browser cached files are loading. Its hard to replicate, but when the issue occurs there are many javascript errors in console. Ads are not loading. If I do cache refresh or ctrl + F5, it works fine. But, I can tell that to website visitors. If I try from different browser, it works without cache refresh. So, I suspect it should be something with browser cached files. It works if I browse the google amp cache url.

Please check below pages.
https://firstranker.com/mbbs/mbbs-2008-2019-question-papers-rajiv-gandhi-university-health-sciences/?amp
https://firstranker.com/fr.php/frdA041119A132440/download-ptu-b.tech-1st-semester-[2019]-70970-engineering-math-iii-question-papers
fr

please find below the errors from console. Attached screenshot
```
amp.js:157 Powered by AMP ⚡ HTML - Version 1911062056110 https://firstranker.com/?amp extensions-impl.js:258 Uncaught (in promise) TypeError: Cannot read property 'setAttribute' of null

at Yl.f.reloadExtension (extensions-impl.js:258)
at jm (runtime.js:919)
at gm (runtime.js:208)
at lm (runtime.js:325)
at wi.qd (amp.js:118)
at vi (chunk.js:192)
at ri.Ze (chunk.js:410)
at chunk.js:461
amp-ad-0.1.js:1 Uncaught TypeError: (intermediate value)(intermediate value).push is not a function

at amp-ad-0.1.js:1
amp-loader-0.1.js:1 Uncaught TypeError: (intermediate value)(intermediate value).push is not a function

at amp-loader-0.1.js:1
amp-auto-lightbox-0.1.js:1 Uncaught TypeError: (intermediate value)(intermediate value).push is not a function

at amp-auto-lightbox-0.1.js:1
```

### PR DESCRIPTION
**Pull request by choumx**
Fixes #25751.

### CODE REVIEW

File: src/service/extension-location.js [hunk 1]
```diff
@@ -98,9 +98,7 @@ export function calculateEntryPointScriptUrl(
  * @return {!ExtensionInfoDef}
  */
 export function parseExtensionUrl(scriptUrl) {
-  const regex = /^(.*)\/(.*)-([0-9.]+)\.js$/i;
-  const matches = scriptUrl.match(regex);
-
+  const matches = scriptUrl.match(/^(.*)\/(.*)-([0-9.]+|latest)\.js$/i);
```
Comment (jridgewell) [inline src/service/extension-location.js]: Nit: We can drop the `^(.*)\/` entirely

Comment (choumx) [reply src/service/extension-location.js]: Don't we at least need the `\/` to separate the extension ID? E.g. `https://cdn.ampproject.org/v0/amp-foo-latest.js`.

File: src/service/extensions-impl.js [hunk 2]
```diff
@@ -271,18 +271,18 @@ export class Extensions {
...
-    // "amp-viewer-integration", do not have "custom-element" attribute.
     const modifier =
       ':not([i-amphtml-loaded-new-version])' +
       (includeInserted ? '' : ':not([i-amphtml-inserted])');
+    // We have to match against "src" because a few extensions, such as
+    // "amp-viewer-integration", do not have "custom-element" attribute.
     const matches = this.win.document.head./*OK*/ querySelectorAll(
       `script[src*="/${extensionId}-"]` + modifier
     );
-    const matcher = new RegExp(`${extensionId}-\\d`);
```
Comment (choumx) [inline src/service/extensions-impl.js]: @gmajoulet This used to match local builds like `amp-foo-0.1.max.js`, but `parseExtensionUrl` does not. Need to update it.

Comment (choumx) [reply src/service/extensions-impl.js]: #25862 should fix this.

**Pull request merged**